tools/oeedger8r/src/Sources.ml

```diff
@@ -178,7 +178,10 @@ let rec get_ptr_setter get_deepcopy args count setter (ptype, decl) =
         (* NOTE: This makes the embedded check in the `OE_` macro superfluous. *)
         [
           sprintf "if (pargs_in->%s)"
-            (String.concat " && pargs_in->" (List.rev (arg :: args)));
+            (if args = [] then arg
+            else ((String.concat " && pargs_in->" (List.rev(args))) ^
+              ((if is_out_setter setter then " && !" else " && ") ^
```

Comment:
I think you can probably just use the logic of `is_out_stter` inline instead of introducing a "common" function that is only used once. But otherwise this looks good to me.

Reply:
Thanks for the suggestion, @andschwa. The patch is updated.